starboard/nplb/posix_compliance/posix_aligned_alloc_test.cc

```diff
@@ -0,0 +1,146 @@
+// limitations under the License.
+
+#include <stdlib.h>
+#include <cerrno>
+#include <cstdint>
+
+#include "testing/gtest/include/gtest/gtest.h"
+
+/*
+ * Untestable Scenarios for aligned_alloc:
+ * - ENOMEM: Reliably triggering out-of-memory conditions in a portable
+ * and non-disruptive way within a unit test is generally not feasible.
+ * - Cases where alignment is not a power of 2, as we check for it in
+ * allocator_shim_default_dispatch_to_partition_alloc.cc and have a compiler
+ * flag that verify this requirement.
+ */
+
+namespace starboard {
+namespace nplb {
+namespace {
+
+// Checks if the given memory address is aligned to the specified alignment.
+bool IsMemoryAligned(const void* ptr, size_t alignment) {
+  if (alignment == 0) {
+    return false;
+  }
+  return reinterpret_cast<uintptr_t>(ptr) % alignment == 0;
+}
+
+TEST(PosixAlignedAllocTests,
+     AllocatesMemorySuccessfullyWhenAlignmentAndSizeAreValid) {
+  // Valid alignment is a power of 2 based on
+  // https://man7.org/linux/man-pages/man3/aligned_alloc.3.html#DESCRIPTION
+  const size_t kValidAlignment = 64;
+  const size_t kValidSize = kValidAlignment * 2;
+  ASSERT_TRUE((kValidAlignment > 0) &&
+              ((kValidAlignment & (kValidAlignment - 1)) == 0));
+  ASSERT_EQ(0, kValidSize % kValidAlignment);
+
+  errno = 0;
+  void* ptr = aligned_alloc(kValidAlignment, kValidSize);
+
+  ASSERT_NE(nullptr, ptr) << "aligned_alloc returned NULL. errno: " << errno;
+  EXPECT_EQ(0, errno);
+  EXPECT_TRUE(IsMemoryAligned(ptr, kValidAlignment));
+
+  free(ptr);
+}
+
+TEST(PosixAlignedAllocTests, HandlesZeroSizeCorrectly) {
+  const size_t kValidAlignment = 128;
+  const size_t kZeroSize = kValidAlignment * 0;
+  ASSERT_TRUE((kValidAlignment > 0) &&
+              ((kValidAlignment & (kValidAlignment - 1)) == 0));
+
+  errno = 0;
+  void* ptr = aligned_alloc(kValidAlignment, kZeroSize);
+
+  // For size 0, aligned_alloc may return NULL or a unique pointer.
+  // It should not set errno to EINVAL if alignment is valid.
+  EXPECT_NE(EINVAL, errno) << "errno was " << errno;
+
+  if (ptr != nullptr) {
+    EXPECT_TRUE(IsMemoryAligned(ptr, kValidAlignment));
+  }
+  free(ptr);
+}
+
+TEST(PosixAlignedAllocTests,
+     AllocatesSuccessfullyWithMinimumValidAlignmentOne) {
+  const size_t kAlignmentOne = 1;  // 2^0 = 1
+  const size_t kAnySize = kAlignmentOne * 100;
+
+  errno = 0;
+  void* ptr = aligned_alloc(kAlignmentOne, kAnySize);
+
+  ASSERT_NE(nullptr, ptr) << "aligned_alloc returned NULL. errno: " << errno;
+  EXPECT_EQ(0, errno);
+  EXPECT_TRUE(IsMemoryAligned(ptr, kAlignmentOne));
+
+  free(ptr);
+}
+
+TEST(PosixAlignedAllocTests,
+     AllocatesSuccessfullyWithAlignmentEqualToSizeofVoidPointer) {
+  const size_t kAlignmentSizeofVoidPtr = sizeof(void*);
+  if (!((kAlignmentSizeofVoidPtr > 0) &&
+        ((kAlignmentSizeofVoidPtr & (kAlignmentSizeofVoidPtr - 1)) == 0))) {
+    GTEST_SKIP() << "sizeof(void*) (" << kAlignmentSizeofVoidPtr
```

Comment:
This _must_ be true, see https://source.corp.google.com/h/github/youtube/cobalt/+/main:starboard/sabi/schema/sabi-v17.schema.json;l=86-90?q=starboard%2Fsabi%2Fschema%2Fsabi-v17.schema.json&ss=h%2Fgithub%2Fyoutube%2Fcobalt%2F%2B%2Frefs%2Fheads%2Fmain. You could change to an ASSERT_TRUE or skip the check, as we already have `starboard/nplb/sabi/alignment_test.cc`